src/__internal__/validations/validation-icon.test.tsx

```diff
@@ -0,0 +1,215 @@
+  await user.hover(validationIcon);
+
+  const tooltip = screen.getByRole("tooltip");
+  expect(tooltip).toBeVisible();
+});
+
+test("renders a tooltip when the validation icon is hovered, then is not rendered when un hovered", async () => {
+  render(<ValidationIcon error="error" />);
+
+  const user = userEvent.setup();
+  const validationIcon = screen.getByTestId("icon-error");
+
+  await user.hover(validationIcon);
+  const tooltip = screen.getByRole("tooltip");
+  expect(tooltip).toBeVisible();
+
+  await user.unhover(validationIcon);
+  expect(tooltip).not.toBeVisible();
+});
+
+test("renders a tooltip when the validation icon is focused", () => {
+  render(<ValidationIcon error="error" />);
+
+  const validationIcon = screen.getByTestId("icon-error");
+  validationIcon.focus();
+
+  const tooltip = screen.getByRole("tooltip");
+  expect(tooltip).toBeVisible();
+});
+
+test("renders a tooltip when validation icon is focused, then is not rendered on blur", async () => {
+  render(<ValidationIcon error="error" />);
+
+  const user = userEvent.setup();
+  const validationIcon = screen.getByTestId("icon-error");
+
+  validationIcon.focus();
+  const tooltip = screen.getByRole("tooltip");
+  expect(tooltip).toBeVisible();
+
+  await user.tab();
+  expect(tooltip).not.toBeVisible();
+});
+
+test("sets the 'id' attribute on the tooltip via the `tooltipId` prop on the validation icon", () => {
+  render(<ValidationIcon tooltipId="foo" error="error" />);
+
+  const validationIcon = screen.getByTestId("icon-error");
+  validationIcon.focus();
+
+  const tooltip = screen.getByRole("tooltip");
+  expect(tooltip).toHaveAttribute("id", "foo");
+});
+
+test("triggers a passed function via the `onFocus` prop when the validation icon is focused", () => {
+  const mockOnFocus = jest.fn();
+  render(<ValidationIcon error="error" onFocus={mockOnFocus} />);
+
+  const validationIcon = screen.getByTestId("icon-error");
+  validationIcon.focus();
+
+  expect(mockOnFocus).toHaveBeenCalled();
+});
+
+test("triggers a passed function via the `onBlur` prop when the validation icon is blurred", async () => {
+  const mockOnBlur = jest.fn();
+  render(<ValidationIcon error="error" onBlur={mockOnBlur} />);
+
+  const user = userEvent.setup();
+  const validationIcon = screen.getByTestId("icon-error");
+
+  await user.click(validationIcon);
+  await user.tab();
+
+  expect(mockOnBlur).toHaveBeenCalled();
+});
+
+test("throws an error when `tooltipFlipOverride` props value is not an array", () => {
+  const mockGlobal = jest
+    .spyOn(global.console, "error")
+    .mockImplementation(() => undefined);
+  const errorMessage = `The tooltipFlipOverrides prop supplied to ValidationIcon must be an array containing some or all of ["top", "bottom", "left", "right"].`;
+
+  expect(() => {
+    render(
+      <ValidationIcon
+        isPartOfInput
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore:next-line
```

Comment:
**suggestion**: what do you think of using [`@ts-expect-error` directive](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html#ts-check) here? This way we can inform the compiler of the expected type error without having to disable the linter: 

```suggestion
       /* @ts-expect-error testing invalid prop value */
```